devito/passes/clusters/implicit.py

```diff
@@ -92,6 +93,8 @@ def callback(self, clusters, prefix):
                 d = None
             if d is None:
                 processed.append(c)
+                # If no MultiSubDomain present in this cluster, then tip should be reset
+                tip = None
```

Comment:
with `thicknesses` now in place, do we actually still need this `tip` thing?

Reply:
Yeah, it keeps track of whether consecutive expressions are on the same SubDomainSet, and avoids generating additional unnecessary thickness expressions in this case. However, it did not previously reset if there was an expression without a SubDomainSet in the middle, which meant that you could create cases where the thickness expressions were erroneously omitted.